exposee-app/Routes/users.js

```diff
@@ -1,18 +1,102 @@
 import express from "express";
 import bcrypt from "bcrypt";
 import { User } from "../models/users.js";
+import { Gift } from "../models/gifts.js";
+import { Wallet } from "../models/wallet.js";
 import { Op } from "sequelize";
 import pkg from "jsonwebtoken";
 const { sign } = pkg;
 import validate_Token from "../authentoken.js";
+import dotenv from "dotenv";
 
 const router = express.Router();
-
+dotenv.config();
+function verifytoken(req, res, next) {
+  const token = req.header("Authorization");
+  if (!token) {
+    return res.status(401).json({ error: "Acess denied. No token provided" });
+  }
+  jwt.verify(token, secretKey, (err, decoded) => {
+    if (err) {
+      return res.status(401).json({ error: "invalid token" });
+    }
+    req.userId = decoded.userId;
+    next();
+  });
+}
 router.get("/user", validate_Token, async (req, res) => {
   const users = await User.findAll();
   console.log(users);
   res.status(200).json(users);
 });
+router.get('/wallet/:user_id', validate_Token, async (req, res) => {
```

Comment:
the restful way of doing this is to do :user_id/wallet, since each user has a wallet. 

If you did /wallet/:wallet_id then that'd be fine.